[JIT] Add GitHub workflow for importing issues to triage project

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#41056 [JIT] Add GitHub workflow for importing issues to triage project**

**Summary**
This commit adds a new GitHub workflow that automatically adds a card to
the "Need triage" section of the project board for tracking JIT triage
for each new issue that is opened and labelled "jit".

**Test Plan**
???

Differential Revision: [D22444262](https://our.internmc.facebook.com/intern/diff/D22444262)

Commit:
[JIT] Add GitHub workflow for importing issues to triage project

**Summary**
This commit adds a new GitHub workflow that automatically adds a card to
the "Need triage" section of the project board for tracking JIT triage
for each new issue that is opened and labelled "jit".

**Test Plan**
???

[ghstack-poisoned]

SplitInfinity commented:
Luckily, GitHub has an action that lets you make GitHub API requests through their Octokit JS client. But I'm not quite sure how to test this without landing?

SplitInfinity commented:
More information on this action: https://github.com/actions/github-script

suo commented on .github/workflows/jit_triage.yml:
```diff
@@ -0,0 +1,72 @@
+
+on:
+  issues:
+    types: [opened]
```
Issue labels are often added after an issue has been opened (either by the pytorch bot or by a human triager). We should cover that case as well.

SplitInfinity replied:
I added `labeled` to the types field, which should handle this. However, it will not cover the case in which the JIT label is removed from an issue and needs to be removed from the project board. Should we do that too?

Maybe later once we get this working.

suo replied:
yeah later is fine

Commit:
Update on "[JIT] Add GitHub workflow for importing issues to triage project"


**Summary**
This commit adds a new GitHub workflow that automatically adds a card to
the "Need triage" section of the project board for tracking JIT triage
for each new issue that is opened and labelled "jit".

**Test Plan**
???

[ghstack-poisoned]

SplitInfinity commented on .github/workflows/jit_triage.yml:
```diff
@@ -0,0 +1,72 @@
+            const filteredProjects = projects.filter(project => project.name == kProjectName);
+
+            if (filteredProjects.length != 1) {
+              console.log("Unable to find a project named " + kProjectName);
```
Good question, it looks like that's what the `core` argument is for: https://github.com/actions/toolkit/tree/master/packages/core

Commit:
Update on "[JIT] Add GitHub workflow for importing issues to triage project"


**Summary**
This commit adds a new GitHub workflow that automatically adds a card to
the "Need triage" section of the project board for tracking JIT triage
for each new issue that is opened and labelled "jit".

**Test Plan**
???

[ghstack-poisoned]

suo commented on .github/workflows/jit_triage.yml:
```diff
@@ -0,0 +1,73 @@
+            const filteredColumns = columns.filter(column => column.name == kColumnName);
+
+            if (filteredColumns != 1) {
+              console.setFailed("Unable to find a column named " + kColumnName);
```
should be `core.setFailed`

suo reviewed (approved):
let's try it

Commit:
Update on "[JIT] Add GitHub workflow for importing issues to triage project"


**Summary**
This commit adds a new GitHub workflow that automatically adds a card to
the "Need triage" section of the project board for tracking JIT triage
for each new issue that is opened and labelled "jit".

**Test Plan**
???

[ghstack-poisoned]
